packages/jsii-dotnet-generator/src/Amazon.JSII.Generator/Class/AbstractClassProxyGenerator.cs

```diff
@@ -0,0 +1,164 @@
+            }
+        }
+
+        protected override IEnumerable<MemberDeclarationSyntax> CreateMethods()
+        {
+            foreach (Method method in GetAllMethods(Type))
+            {
+                var generator = new AbstractClassProxyMethodGenerator(Type, method, Symbols, Namespaces);
+                yield return generator.CreateMethod();
+            }
+        }
+
+        private IEnumerable<Method> GetAllMethods(Type type)
+        {
+            IEnumerable<Method> GetAllMethodsRecurse(Type currentType, IEnumerable<Method> methods)
+            {
+                if (currentType is InterfaceType interfaceType)
+                {
+                    // Get all properties from the interface.
+                    methods = methods.Concat(interfaceType.Methods ?? Enumerable.Empty<Method>());
+
+                    // Interfaces can have superinterfaces. Run through them too.
+                    if (interfaceType.Interfaces != null)
+                    {
+                        var superinterfaceMethods = interfaceType.Interfaces.Select(r =>
+                                Symbols.GetTypeFromFullyQualifiedName(r.FullyQualifiedName) as
+                                    InterfaceType)
+                            .SelectMany(i => GetAllMethodsRecurse(i, methods))
+                            .ToList();
+
+                        methods = methods.Concat(superinterfaceMethods);
+                    }
+                }
+                else if (currentType is ClassType classType)
+                {
+                    // Get all methods from the interface
+                    methods = methods.Concat(classType.Methods ?? Enumerable.Empty<Method>());
+
+                    if (classType.Interfaces != null)
+                    {
+                        // Run through all the interfaces.
+                        var superinterfaceMethods = classType.Interfaces.Select(r =>
+                                Symbols.GetTypeFromFullyQualifiedName(r.FullyQualifiedName) as
+                                    InterfaceType)
+                            .SelectMany(i => GetAllMethodsRecurse(i, methods))
+                            .ToList();
+
+                        methods = methods.Concat(superinterfaceMethods);
+                    }
+
+                    // Run through the superclass.
+                    if (classType.Base != null)
+                    {
+                        methods = methods.Concat(GetAllMethodsRecurse(
+                            Symbols.GetTypeFromFullyQualifiedName(classType.Base.FullyQualifiedName) as ClassType,
+                            methods));
+                    }
+                }
+
+                return methods;
+            }
+
+            // Only get the first declaration encountered, and keep it if it is abstract.
+            return GetAllMethodsRecurse(type, Enumerable.Empty<Method>())
+                .GroupBy(m => (m.Name,
```

Comment:
What's the point of this `GroupBy`? Can you add a comment explaining?

Reply:
I can add a comment.

Reply:
Actually, I already did. The list contains ALL methods and properties encountered, in the order encountered. An abstract class can have concrete implementations. Therefore, we only generate methods/properties if the first member encountered is unimplemented.